index.js

```diff
@@ -118,8 +118,95 @@ async function getAllDatasets(dataCatalogUrl = 'https://openactive.io/data-catal
   return { jsonld: jsonldFromDatasetUrls, errors };
 }
 
+/**
+ * Validates JSON-LD content by ensuring the '@id' or 'id' field matches the provided ID.
+ *
+ * This function performs an HTTP GET request to the specified ID (URL), retrieves
+ * the response, and extracts JSON-LD from it if needed and possible. It then compares
+ * the '@id' or 'id' field from the retrieved JSON-LD to the provided ID.
+ *
+ * Note that this is only applicable to JSON-LD "@id" for the DataCatalog and Dataset types, which must resolve.
+ *
+ * @async
+ * @function validateJsonLdId
+ * @param {string} id - The expected '@id' or 'id' value, also the URL to be requested.
+ * @param {boolean} expectHtml - A flag indicating whether the response is expected to be HTML (i.e. a Dataset Site).
+ * @returns {Promise<{isValid: boolean, error: string|null}>} - An object indicating the validity
+ * of the JSON-LD and any associated error message.
+ *
+ * @example
+ * validateJsonLdId('https://example.com/data.jsonld', false)
+ *   .then(({isValid, error}) => {
+ *     if (isValid) {
+ *       console.log('JSON-LD is valid!');
+ *     } else {
+ *       console.error(`JSON-LD validation failed: ${error}`);
+ *     }
+ *   });
+ */
+async function validateJsonLdId(id, expectHtml) {
+  let response;
+
+  try {
+    response = await axiosGetWithRetry(id);
+    response = response.data;
+  } catch (error) {
+    return { isValid: false, error: `Failed to resolve URL: ${error.message}` };
+  }
+
+  let jsonLd;
+  try {
+    if (expectHtml && typeof response === 'string') {
+      jsonLd = extractJSONLDfromHTML(id, response);
+    } else if (!expectHtml && typeof response === 'object') {
+      jsonLd = response;
+    } else {
+      return { isValid: false, error: `Unexpected response type: ${typeof response}` };
+    }
+
+    const jsonId = jsonLd['@id'] || jsonLd.id;
+    if (jsonId !== id) {
+      return { isValid: false, error: `Mismatched '@id': From file: "${id}"; From referenced JSON-LD: "${jsonId}"` };
+    }
+  } catch (error) {
+    return { isValid: false, error: error.message };
+  }
+
+  return { isValid: true, error: null };
+}
+
+/*
+* System-specific workaround: Note that rate limits in Legend can cause this request to fail with a 403 (?), so we retry up to 5 times
+* TODO: Ask Legend to return a 429 instead
+*/
+async function axiosGetWithRetry(url) {
```

Comment:
This is a super-minor suggestion so up to you but maybe rename to `axiosGetWithRetryForKnownLegendIssue`?